lib/IMAP/FolderMapper.php

```diff
@@ -76,8 +86,17 @@ public function getFolders(Account $account, Horde_Imap_Client_Socket $client,
 
 	public function createFolder(Horde_Imap_Client_Socket $client,
 		Account $account,
-		string $name): Folder {
-		$client->createMailbox($name);
+		string $name,
+		array $specialUseAttributes = []): Folder {
+		$opts = [];
+	
+		foreach ($specialUseAttributes as $attribute) {
+			if (!in_array($attribute, self::SUPPORTED_SPECIAL_USE_ATTRIBUTES, true)) {
+				throw new ServiceException('Unsupported special use attribute: ' . $attribute);
+			}
+			$opts[] = $attribute;
+		}
+		$client->createMailbox($name, $opts);
```

Comment:
createMailbox expects an array "opts" with a key "special_use" and there the list of attributes.